cmd/up.go

```diff
@@ -142,3 +165,46 @@ func isGKE(disco discovery.DiscoveryInterface) (bool, error) {
 
 	return false, nil
 }
+
+func mongoSecretExists(c kubecfg.ApplyCmd, name, ns string) (*unstructured.Unstructured, bool, error) {
+	gvk := schema.GroupVersionKind{Version: "v1", Kind: "Secret"}
+	rc, err := clientForGroupVersionKind(c.ClientPool, c.Discovery, gvk, ns)
+	if err != nil {
+		return nil, false, err
+	}
+	pwFields := []string{"mongodb-password", "mongodb-root-password"}
+	prevSec, err := rc.Get(name)
+	if err != nil {
+		if k8sErrors.IsNotFound(err) {
+			return nil, false, nil
+		} else {
+			return nil, true, err
+		}
+	} else {
```

Comment:
Remove nested `else` clauses when all the `if` codepaths exit anyway - it reads more easily.
(Here, and a much smaller version 2 lines above)